### PR TITLE
Messaging: Remove PublisherTrigger

### PR DESCRIPTION
**Pull request by dstenroejl**
## Description

Remove unwise dependency to `Energinet.DataHub.Core.App.WebApp`.

Remove `PublisherTrigger` and related types and DI extensions because the trigger was based on a type from the App packages that was also removed.

## Quality

- [x] Documentation is updated
- [x] Release notes are updated
- [x] Package version is updated
- [ ] Public types and methods are documented
- [ ] Tests are implemented and executed locally

### CODE REVIEW

File: source/Messaging/documents/documentation.md [hunk 1]
```diff
@@ -19,8 +19,7 @@ The package is still work in progress.
 
 ## Publishing
 
-The publishing functionality is responsible for publishing integration events. It can be used manually or through a hosted service.
-Regardless of which "mode" is used, the IIntegrationEventProvider implementation has to be implemented.
+The publishing functionality is responsible for publishing integration events. The IIntegrationEventProvider implementation has to be implemented.
```
Comment (jonasdmoeller) [inline source/Messaging/documents/documentation.md:22]: Maybe: "The IIntegrationEventProvider interface has to be implemented.", as it is an interface more than an implementation.

File: source/Messaging/documents/documentation.md [hunk 2]
```diff
@@ -60,20 +59,9 @@ services.Configure<PublisherOptions>(builder.Configuration.GetSection(nameof(Pub
 services.AddPublisher<IntegrationEventProvider>();
 ```
 
-### Manual
-
-When publishing manually, the above IIntegrationEventProvider implementation and registration is enough to start publishing integration events.
+When publishing, the above IIntegrationEventProvider implementation and registration is enough to start publishing integration events.
```
Comment (jonasdmoeller) [inline source/Messaging/documents/documentation.md:62]: Same as above

File: source/Messaging/documents/documentation.md (hunk 1 above)
Comment (dstenroejl) [reply source/Messaging/documents/documentation.md]: Fixed

File: source/Messaging/documents/documentation.md (hunk 2 above)
Comment (dstenroejl) [reply source/Messaging/documents/documentation.md]: Fixed

**Review by jonasdmoeller — APPROVED** [review]

**Pull request merged**